xcodeuitest/xcodeuitestcmd.go

```diff
@@ -0,0 +1,255 @@
+	// For more info about the possible values please see xcodebuild's docs about the -sdk flag.
+	// Only passed to xcodebuild if not empty!
+	SDK string
+}
+
+// RunBuildForTesting runs the build-for-tesing xcode command
+func (xcuitestcmd CommandModel) RunBuildForTesting() (string, string, error) {
+	xcoutput := ""
+	var err error
+
+	tmpDir, err := pathutil.NormalizedOSTempDirPath("__codesigndoc__")
+	if err != nil {
+		return "", "", fmt.Errorf("failed to create temp dir for archives, error: %s", err)
+	}
+	tmpBuildPath := filepath.Join(tmpDir, xcuitestcmd.Scheme)
+
+	progress.SimpleProgress(".", 1*time.Second, func() {
+		xcoutput, err = xcuitestcmd.RunXcodebuildCommand("clean", "build-for-testing", "CONFIGURATION_BUILD_DIR="+tmpBuildPath)
+	})
+	fmt.Println()
+
+	if err != nil {
+		return "", xcoutput, err
+	}
+	return tmpBuildPath, xcoutput, nil
+}
+
+func (xcuitestcmd CommandModel) xcodeProjectOrWorkspaceParam() (string, error) {
+	if strings.HasSuffix(xcuitestcmd.ProjectFilePath, "xcworkspace") {
+		return "-workspace", nil
+	} else if strings.HasSuffix(xcuitestcmd.ProjectFilePath, "xcodeproj") {
+		return "-project", nil
+	}
+	return "", fmt.Errorf("invalid project/workspace file, the extension should be either .xcworkspace or .xcodeproj ; (file path: %s)", xcuitestcmd.ProjectFilePath)
+}
+
+func (xcuitestcmd CommandModel) transformToXcodebuildParams(xcodebuildActionArgs ...string) ([]string, error) {
+	projParam, err := xcuitestcmd.xcodeProjectOrWorkspaceParam()
+	if err != nil {
+		return []string{}, err
+	}
+
+	baseArgs := []string{projParam, xcuitestcmd.ProjectFilePath}
+	if xcuitestcmd.Scheme != "" {
+		baseArgs = append(baseArgs, "-scheme", xcuitestcmd.Scheme)
+	}
+
+	if xcuitestcmd.SDK != "" {
+		baseArgs = append(baseArgs, "-sdk", xcuitestcmd.SDK)
+	}
+
+	return append(baseArgs, xcodebuildActionArgs...), nil
+}
+
+// RunXcodebuildCommand TODO comment
+func (xcuitestcmd CommandModel) RunXcodebuildCommand(xcodebuildActionArgs ...string) (string, error) {
+	xcodeCmdParamsToRun, err := xcuitestcmd.transformToXcodebuildParams(xcodebuildActionArgs...)
+	if err != nil {
+		return "", err
+	}
+
+	log.Infof("$ xcodebuild %s", command.PrintableCommandArgs(true, xcodeCmdParamsToRun))
+	xcoutput, err := command.RunCommandAndReturnCombinedStdoutAndStderr("xcodebuild", xcodeCmdParamsToRun...)
+	if err != nil {
+		return xcoutput, fmt.Errorf("failed to run xcodebuild command, error: %s", err)
+	}
+
+	log.Debugf("xcoutput: %s", xcoutput)
+	return xcoutput, nil
+}
+
+// ScanSchemes scans the provided project or workspace for schemes.
+// Returns the schemes of the provided project's or the provided workspace's + project's schemes, the names of the schemes,
+// the schemes with UITest target and the schemes' which has UITest.
+func (xcuitestcmd CommandModel) ScanSchemes() (schemes []xcscheme.Scheme, schemesWitUITests []xcscheme.Scheme, schemeNames []string, schemesWitUITestNames []string, err error) {
```

Comment:
the `...Names` return values are redundant as the `scheme...` return values are holding this information

Reply:
👍